src/modules/product-category/product-category.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, NotFoundException, BadRequestException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { ProductCategory } from './entities/product-category.entity';
+import { CreateProductCategoryDto } from './dto/create-product-category.dto';
+import { UpdateProductCategoryDto } from './dto/update-product-category.dto';
+
+@Injectable()
+export class ProductCategoryService {
+  constructor(
+    @InjectRepository(ProductCategory)
+    private categoryRepository: Repository<ProductCategory>
+  ) {}
+
+  async createCategory(createCategoryDto: CreateProductCategoryDto): Promise<ProductCategory> {
+    const category = this.categoryRepository.create(createCategoryDto);
+    return await this.categoryRepository.save(category);
+  }
+
+  async getAllCategories(limit?: number, offset?: number): Promise<ProductCategory[]> {
+    if (limit !== undefined && (isNaN(Number(limit)) || Number(limit) < 0)) {
```

Comment:
I need you to review this your condition here, it seems off

Reply:
okay. i didn't write that part of the code, i only did the post endpoint for category because i need it for my endpoint. so can i comment that out in my repo